packages/astro/src/core/ignore-files.ts

```diff
@@ -0,0 +1,28 @@
+import fs from 'node:fs';
+import { joinPaths } from './path.js';
+
+/**
+ * use .gitignore file to ignore files
+ * @param root astro root path
+ */
+export async function ignoreFiles(root: string) {
+	const res: string[] = [];
+	const gitIgnoreFile = joinPaths(root, '.gitignore');
+	const text = await fs.promises
+		.readFile(gitIgnoreFile, 'utf-8')
+		.then((data) => data.toString())
+		.catch(() => '');
+	
+	if (text.length > 0) {
+		const lines = text.split('\n');
```

Comment:
In windows, you might not have `\n` but you have `\r` or `\r\n`. Use [EOL](https://nodejs.org/dist/latest-v20.x/docs/api/os.html#oseol) instead